relay/sources/relayd/src/shared_files.rs

```diff
@@ -119,23 +158,36 @@ pub fn parse_value(key: &str, file: &str) -> Result<String, ()> {
 }
 
 pub fn same_hash_than_in_nodeslist(
-    hash_value: &str,
-    source_uuid: String,
-    job_config: Arc<JobConfig>,
-) -> bool {
-    let myvec2: Vec<String> = job_config
-        .nodes
-        .read()
-        .expect("Cannot read nodes list")
-        .get_keyhash_from_uuid(&source_uuid)
-        .unwrap()
-        .split(':')
-        .map(|s| s.to_string())
-        .collect();
+    pubkey: String,
+    hash_type: String,
+    keyhash: String,
+) -> Result<bool, Error> {
+    let pubkey_pem = get_pubkey(pubkey)?;
 
-    let (_algorithm, keyhash) = (&myvec2[0], &myvec2[1]);
+    let public_key_der = pubkey_pem.public_key_to_der()?;
 
-    hash_value == keyhash
+    let bytes2: &[u8] = &public_key_der;
```

Comment:
`bytes2` is a strange name, but you can probably avoid it with the `as_bytes()` method.